### PR TITLE
feat: compute contract address

### PR DESCRIPTION
**Pull request by enitrat**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #562

## What is the new behavior?

- compute contract address
- improved namings of some variables
- moved functions in appropriate modules

## Does this introduce a breaking change?

- [ ] Yes
- [n] No

### CODE REVIEW

File: crates/evm/src/tests/test_model.cairo [hunk 1]
```diff
@@ -119,21 +119,21 @@ fn test_account_is_deployed_contract_account() {
...
     // Then
-    assert(account.is_deployed() == true, 'account should be deployed');
+    assert(account.exists() == true, 'account should be deployed');
 }
 
 
 #[test]
 #[available_gas(5000000)]
-fn test_account_is_deployed_undeployed() {
+fn test_account_exists_undeployed() {
```
Comment (Eikix) [inline crates/evm/src/tests/test_model.cairo:128]: we're not checking the last case no?
the case where a contract is scheduled for deployment, but isn't deployed yet should have:
- exists() == true
- is_deployed() == false

File: crates/utils/src/rlp.cairo [hunk 2]
```diff
@@ -64,63 +64,73 @@ impl RLPImpl of RLPTrait {
...
+                },
+                Option::None => { break; }
+            }
+        };
+        let len_joined_encodings = joined_encodings.len();
+        if len_joined_encodings < 0x38 {
+            let mut result: Array<u8> = array![0xC0 + len_joined_encodings.try_into().unwrap()];
+            result.append_span(joined_encodings.span());
+            return result.span();
+        } else {
```
Comment (Eikix) [inline crates/utils/src/rlp.cairo:94]: can this else case even happen?

**Review by Eikix — APPROVED** [review]
lgtm

File: crates/utils/src/rlp.cairo (hunk 2 above)
Comment (enitrat) [reply crates/utils/src/rlp.cairo:94]: normally we're never going to reach this code - but that's how the encoding of sequences of strings should be done.

File: crates/evm/src/create_helpers.cairo [hunk 3]
```diff
@@ -48,15 +49,15 @@ impl MachineCreateHelpersImpl of MachineCreateHelpers {
 
         // TODO(state): when the tx starts,
         // store get_tx_info().unbox().nonce inside the sender account nonce
-        let sender_nonce = get_tx_info().unbox().nonce;
+        let sender_nonce: u64 = get_tx_info().unbox().nonce.try_into().unwrap();
```
Comment (Eikix) [inline crates/evm/src/create_helpers.cairo:52]: I think it'd be cleaner to only get the nonce for the CreateOrDeployTx case

File: crates/evm/src/create_helpers.cairo [hunk 4]
```diff
@@ -49,12 +49,13 @@ impl MachineCreateHelpersImpl of MachineCreateHelpers {
 
         // TODO(state): when the tx starts,
         // store get_tx_info().unbox().nonce inside the sender account nonce
-        let sender_nonce: u64 = get_tx_info().unbox().nonce.try_into().unwrap();
+        // so that we can call self.nonce() instead of get_tx_info().unbox().nonce
```
Comment (Eikix) [inline crates/evm/src/create_helpers.cairo:52]: Or add it as a tasklist for deploy tx PR? if the issue doesnt exist yet for tracking

Comment (enitrat) [reply crates/evm/src/create_helpers.cairo:52]: ok

**Review by Eikix — APPROVED** [review]
lgtm

**Pull request merged**